src/site/content/en/blog/extending-workbox/index.md

```diff
@@ -0,0 +1,379 @@
+strategy (which itself extends the `Strategy` base):
+
+```javascript
+// See https://developers.google.com/web/tools/workbox/guides/using-bundlers
+import {NetworkFirst} from 'workbox-strategies';
+
+class DedupeNetworkFirst extends NetworkFirst {
+  constructor(options) {
+    super(options);
+    // This maps inflight requests to response promises.
+    this._requests = new Map();
+  }
+
+  // _handle is the standard entry point for our logic.
+  async _handle(request, handler) {
+    let responsePromise = this._requests.get(request.url);
+
+    if (responsePromise) {
+      // If there's already an inflight request, return a copy
+      // of the eventual response.
+      const response = await responsePromise;
+      return response.clone();
+    } else {
+      // If there isn't already an inflight request, then use
+      // the _handle() method of NetworkFirst to kick one off.
+      responsePromise = super._handle(request, handler);
+      this._requests.set(request.url, responsePromise);
+      try {
+        const response = await responsePromise;
+        return response.clone();
+      } finally {
+        // Make sure to clean up after a batch of inflight
+        // requests are fulfilled!
+        this._requests.delete(request.url);
+      }
+    }
+  }
+}
+```
+
+{% Aside %}
+This code assumes that all requests for the same URL can be
+satisfied with the same response, which won't always be the case if cookies or
+session state information comes into play.
+{% endAside %}
+
+### CacheNetworkRace
+
+Here's another example of a custom strategy—one that's a twist on
+stale-while-revalidate, where both the network and cache are checked at the same
+time, with a race to see which will return a response first.
+
+```javascript
+// See https://developers.google.com/web/tools/workbox/guides/using-bundlers
+import {Strategy} from 'workbox-strategies';
+
+// Instead of extending an existing strategy,
+// this extends the generic Strategy base class.
+class CacheNetworkRace extends Strategy {
+  // _handle is the standard entry point for our logic.
+  _handle(request, handler) {
+    // handler is an instance of the StrategyHandler class,
+    // and exposes helper methods for interacting with the
+    // cache and network.
+    const fetchDone = handler.fetchAndCachePut(request);
+    const matchDone = handler.cacheMatch(request);
+
+    // The actual response generation logic relies on a "race"
+    // between the network and cache promises.
+    return new Promise((resolve, reject) => {
+      fetchDone.then(resolve);
+      matchDone.then((response) => response && resolve(response));
+
+      // Promise.allSettled() is implemented in recent browsers.
+      Promise.allSettled([fetchDone, matchDone]).then(results => {
+        if (results[0].status === 'rejected' &&
+            !results[1].value) {
+          reject(results[0].reason);
+        }  
+      });
+    });
+  }
+}
+```
+
+## Using the StategyHandler
```

Comment:
```suggestion
## StategyHandler: the recommended approach for creating custom strategies
```
`Using the StrategyHandler` doesn't give me a clue about why I would want to use it